webapp/app/js/compare.js

```diff
@@ -0,0 +1,448 @@
+             cmap['originalVariation'] = "+/- " + rawResultsMap[seriesList[idx].signature].variation.toFixed(2);
+          }
+          if (new_seriesList[idx].signature in new_rawResultsMap) {
+             cmap['newGeoMean'] = new_rawResultsMap[new_seriesList[idx].signature].geomean.toFixed(2);
+             cmap['newRuns'] = new_rawResultsMap[new_seriesList[idx].signature].runs;
+//TODO: \b1 doesn't print out the +- character
+//                     cmap['newVariation'] = "\B1" + rawResultsMap[seriesList[idx].signature].variation
+             cmap['newVariation'] = "+/- " + rawResultsMap[new_seriesList[idx].signature].variation.toFixed(2);
+          }
+
+          if (new_seriesList[idx].signature in new_rawResultsMap && 
+              seriesList[idx].signature in rawResultsMap) {
+             cmap['delta'] = (cmap['newGeoMean'] - cmap['originalGeoMean']).toFixed(2);
+             cmap['deltaPercentage'] = (cmap['delta'] / cmap['originalGeoMean'] * 100).toFixed(2) + ' %';
+
+            cmap.type = 'data';
+            //TODO: some tests are reverse, figure out how to account for that
+            if (cmap.delta < 0) {
+              cmap.type = 'improvement';
+            } else if (cmap.delta > 1) {
+              cmap.type = 'regression';
+            }
+
+            //TODO: zoom?  >1 highlighted revision?
+            var originalSeries = encodeURIComponent(JSON.stringify(
+                          { project: $scope.originalProject,
+                            signature: seriesList[idx].signature,
+                            visible: true}));
+
+            var newSeries = encodeURIComponent(JSON.stringify(
+                          { project: $scope.newProject,
+                            signature: new_seriesList[idx].signature,
+                            visible: true}));
+
+            var detailsLink = thServiceDomain + '/perf.html#/graphs?timerange=' +
+                $scope.timeRange + '&series=' + newSeries;
+
+            if (seriesList[idx].signature != new_seriesList[idx].signature) {
+              detailsLink += '&series=' + originalSeries;
+            }
+
+            detailsLink += '&highlightedRevision=' + $scope.newRevision;
+
+            cmap.detailsLink = detailsLink;
+            cmap.name = seriesList[idx].platform;
+            cmap.isHeader = false;
+            compareResultsMap[counter++] = cmap;
+          }
+
+        };
+      });
+      $scope.compareResults = Object.keys(compareResultsMap).map(function(k) {
+        return compareResultsMap[k];
+      });
+    }
+
+    function verifyRevision(project, revision, rsid) {
+      if ((revision != null && revision != '') &&
+          (project != null && project != '')) {
+
+        var uri = thServiceDomain + '/api/project/' + project +
+            '/resultset/?format=json&full=false&with_jobs=false&revision=' + 
+            revision;
+
+        $http.get(uri).then(function(response) {
```

Comment:
To get a promise here, you can just put a return statement in front of $http.get